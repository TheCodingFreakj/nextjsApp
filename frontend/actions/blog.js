import axios from "axios";
import { API } from "../config";

//This is the input I am getting from the createBlogForm() after submit
//formData, getCookie("token")
export const createBlog = async (blog, token) => {
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        Authorization: ` Bearer ${token}`,
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const body = blog; //we are sending the formData to get posted on the backend

    const response = await axios.post(`${API}/api/blog`, body, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

export const listBlogsWithCategoriesNTags = async () => {
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    // const body = blog;

    const response = await axios.post(
      `${API}/api/blogs-categories-tags`,
      config
    ); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};
