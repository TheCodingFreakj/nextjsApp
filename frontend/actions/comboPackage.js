import axios from "axios";
import { API } from "../config";

export const createNewComboPackage = async (comboPackage, token) => {
  console.log("This is the data I am sending to server", comboPackage);

  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        Authorization: ` Bearer ${token}`,
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const body = comboPackage;
    //console.log("The body is", body);
    const response = await axios.post(`${API}/api/comboPackage`, body, config);
    //console.log(response);
    return response.data;
    console.log(response.data);
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

export const createNewComboPackagePrice = async (comboPackagePrice, token) => {
  //console.log("This is the data I am sending to server", serviceInfo);
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        Authorization: ` Bearer ${token}`,
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const body = comboPackagePrice; //we are sending the formData to get posted on the backend

    const response = await axios.post(
      `${API}/api/combo-package-price`,
      body,
      config
    ); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

//get the tool info and client price to display
export const getAllPackagePriceOptions = async () => {
  try {
    const config = {
      method: "GET",
    };

    const response = await axios.get(`${API}/api/get-combo-price`, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      console.log("Error", error.message);
    }
  }
};

//get the tool info and client price to display
export const getComboPackages = async () => {
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };
    const response = await axios.post(`${API}/api/all-combo-Packages`, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      console.log("Error", error.message);
    }
  }
};
