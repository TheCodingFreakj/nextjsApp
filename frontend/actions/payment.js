import axios from "axios";
import { API } from "../config";

export const createSubsription = async (paymentData, token) => {
  console.log(paymentData);
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        Authorization: ` Bearer ${token}`,
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const body = JSON.stringify(paymentData);
    console.log(body);

    const response = await axios.post(
      `${API}/api/subscribe-services`,
      body,
      config
    ); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      console.log("Error", error.message);
    }
  }
};

// export const getCategories = async () => {
//   try {
//     const config = {
//       method: "GET",
//     };

//     const response = await axios.get(`${API}/api/categories`, config); //handing the backedn register user
//     return response.data;
//     console.log(response.data); // this is the token from backend
//     console.log(response.status);
//     console.log(response.statusText);
//     console.log(response.headers);
//     console.log(response.config);
//     //we can use this token to access protected routes
//   } catch (error) {
//     if (error.response) {
//       // The request was made and the server responded with a status code
//       // that falls out of the range of 2xx
//       console.log(error.response.data);
//       console.log(error.response.status);
//       console.log(error.response.headers);
//     } else {
//       // Something happened in setting up the request that triggered an Error
//       console.log("Error", error.message);
//     }
//   }
// };
