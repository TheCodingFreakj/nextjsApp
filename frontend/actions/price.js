// import axios from "axios";
// import { API } from "../config";

// export const createNewPrice = async (newPriceInfo, token) => {
//   try {
//     const config = {
//       method: "POST",
//       headers: {
//         Accept: "application/json",
//         Authorization: ` Bearer ${token}`,
//         "Content-Type": "application/json",
//         "Access-Control-Allow-Origin": "*",
//         "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
//       },
//     };

//     const body = JSON.stringify(newPriceInfo);
//     console.log(body);
//     //Fine;

//     const response = await axios.post(`${API}/api/createPrice`, body, config); //handing the backedn register user
//     return response.data;
//     console.log(response.data); // this is the token from backend
//     console.log(response.status);
//     console.log(response.statusText);
//     console.log(response.headers);
//     console.log(response.config);
//     //we can use this token to access protected routes
//   } catch (error) {
//     if (error.response) {
//       // The request was made and the server responded with a status code
//       // that falls out of the range of 2xx
//       console.log(error.response.data);
//       console.log(error.response.status);
//       console.log(error.response.headers);
//     } else {
//       // Something happened in setting up the request that triggered an Error
//       console.log("Error", error.message);
//     }
//   }
// };

// //get the tool info and client price to display
// export const getAllServicesOptions = async (token) => {
//   try {
//     const config = {
//       method: "GET",
//       headers: {
//         Accept: "application/json",
//         Authorization: ` Bearer ${token}`,
//         "Content-Type": "application/json",
//         "Access-Control-Allow-Origin": "*",
//         "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
//       },
//     };

//     const response = await axios.get(`${API}/api/get-price-list`, config); //handing the backedn register user
//     return response.data;
//     console.log(response.data); // this is the token from backend
//     console.log(response.status);
//     console.log(response.statusText);
//     console.log(response.headers);
//     console.log(response.config);
//     //we can use this token to access protected routes
//   } catch (error) {
//     if (error.response) {
//       // The request was made and the server responded with a status code
//       // that falls out of the range of 2xx
//       console.log(error.response.data);
//       console.log(error.response.status);
//       console.log(error.response.headers);
//     } else {
//       // Something happened in setting up the request that triggered an Error
//       console.log("Error", error.message);
//     }
//   }
// };
