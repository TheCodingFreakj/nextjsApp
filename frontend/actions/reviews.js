import axios from "axios";
import { API } from "../config";

export const createNewReview = async (dataForBackened, token) => {
  console.log("This is the data I am sending to server", dataForBackened);

  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        Authorization: ` Bearer ${token}`,
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const body = dataForBackened; //we are sending the formData to get posted on the backend

    const response = await axios.post(
      `${API}/api/create-reviews`,
      body,
      config
    ); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

export const getAllBrands = async () => {
  try {
    const config = {
      method: "GET",
    };

    const response = await axios.get(`${API}/api/getBrands`, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};
