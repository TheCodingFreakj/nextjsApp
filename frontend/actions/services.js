import axios from "axios";
import { API } from "../config";

export const createServices = async (serviceInfo, token) => {
  console.log("This is the data I am sending to server", serviceInfo);

  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        Authorization: ` Bearer ${token}`,
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const body = serviceInfo; //we are sending the formData to get posted on the backend

    const response = await axios.post(`${API}/api/services`, body, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

export const getDiscountedPrice = async () => {
  try {
    const config = {
      method: "GET",
    };

    const response = await axios.get(`${API}/api/services`, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

//get the tool info and client price to display
export const getAllServices = async () => {
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };
    const response = await axios.post(`${API}/api/all-services`, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      console.log("Error", error.message);
    }
  }
};

// export const getSingleService = async (params) => {
//   try {
//     const config = {
//       method: "GET",
//     };

//     const response = await axios.get(`${API}/api/services`, config); //handing the backedn register user
//     return response.data;
//     console.log(response.data); // this is the token from backend
//     console.log(response.status);
//     console.log(response.statusText);
//     console.log(response.headers);
//     console.log(response.config);
//     //we can use this token to access protected routes
//   } catch (error) {
//     if (error.response) {
//       // The request was made and the server responded with a status code
//       // that falls out of the range of 2xx

//       return error.response.data;
//       console.log(error.response.data);
//       console.log(error.response.status);
//       console.log(error.response.headers);
//     } else {
//       // Something happened in setting up the request that triggered an Error
//       return error.message;
//       console.log("Error", error.message);
//     }
//   }
// };

export const updateService = async (slug, token) => {
  try {
    const config = {
      method: "PUT",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        Authorization: ` Bearer ${token}`,
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const response = await axios.get(`${API}/api/services/${slug}`, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

export const removeService = async (slug, token) => {
  try {
    const config = {
      method: "DELETE",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        Authorization: ` Bearer ${token}`,
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const response = await axios.get(`${API}/api/services/${slug}`, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx

      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};
