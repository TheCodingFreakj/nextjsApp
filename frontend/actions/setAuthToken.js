import cookie from "js-cookie";

export const setCookie = (key, value) => {
  if (typeof window !== "undefined") {
    cookie.set(key, value, {
      expires: 1,
    });
  }
};

export const removeCookie = (key) => {
  if (typeof window !== "undefined") {
    cookie.remove(key, {
      expires: 1,
    });
  }
};

export const getCookie = (key) => {
  if (typeof window !== "undefined") {
    return cookie.get(key);
  }
};

//key is the name of the storage
export const setLocatStorage = (key, value) => {
  if (typeof window !== "undefined") {
    localStorage.setItem(key, JSON.stringify(value));
  }
};

export const removeLocatStorage = (key, value) => {
  if (typeof window !== "undefined") {
    localStorage.removeItem(key);
  }
};

//authenticate user by passing the data what you got got in post login axios req while signin
export const authenticate = (data, next) => {
  setCookie("token", data.token);
  setLocatStorage("user", data.user);
  next();
};

export const isAuth = () => {
  if (typeof window !== "undefined") {
    const cookiesChecked = getCookie("token");

    if (cookiesChecked) {
      if (localStorage.getItem("user")) {
        try {
          return JSON.parse(localStorage.getItem("user"));
        } catch (error) {
          console.log(error);
          console.error("Not a JSON response");
        }
      } else {
        return false;
      }
    }
  }
};

export const userRole = () => {
  if (typeof window !== "undefined") {
    const cookiesChecked = getCookie("token");
    if (cookiesChecked) {
      if (localStorage.getItem("user")) {
        try {
          const user = JSON.parse(localStorage.getItem("user"));
          if (user.role === 1 || user.role === 0) {
            return user.role;
          } else if (user.customerRole === "consumer") {
            return user.customerRole;
          }
          // console.log("the user", user);
        } catch (error) {
          console.log(error);
          console.error("Not a JSON response");
        }

        //return JSON.parse(localStorage.getItem("user") || "");
      } else {
        return false;
      }
    }
  }
};

export const updateUserLocalStorage = (user, next) => {
  if (typeof window !== "undefined") {
    if (localStorage.getItem("user")) {
      let auth = JSON.parse(localStorage.getItem("user"));
      auth = user;
      localStorage.setItem("user", JSON.stringify(auth));
    }
  }
};
