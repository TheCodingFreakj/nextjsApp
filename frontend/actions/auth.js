//import fetch from "isomorphic-fetch";
import axios from "axios";
import { API } from "../config";
import { removeCookie, removeLocatStorage } from "./setAuthToken";
//import setAuthTokenLocalStorage from "../actions/setAuthTokenLocalStorage";

export const signup = async (newUser) => {
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };

    const body = JSON.stringify(newUser);

    const response = await axios.post(`${API}/api/sign-up`, body, config); //handing the backedn register user
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error

      return error.message;
      console.log("Error", error.message);
    }
  }
};

//Load The User

export const signin = async (user) => {
  try {
    const config = {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET,PUT,POST,DELETE,PATCH,OPTIONS",
      },
    };
    const body = JSON.stringify(user);
    const response = await axios.post(
      "http://localhost:8000/api/sign-in",
      body,
      config
    );
    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
    //we can use this token to access protected routes
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};

export const signout = async (next) => {
  removeCookie("token");
  removeLocatStorage("user");

  next();

  const config = {
    method: "GET",
  };

  try {
    await axios.get(`${API}/api/sign-out`, config);
    console.log("Signout successful");
  } catch (error) {
    console.log("Error", error.message);
  }
};

export const authencticateUser = async (token) => {
  const config = {
    method: "GET",
  };

  try {
    await axios.get(`${API}/api/sign-in`, config, token);
    console.log("update successful");

    return response.data;
    console.log(response.data); // this is the token from backend
    console.log(response.status);
    console.log(response.statusText);
    console.log(response.headers);
    console.log(response.config);
  } catch (error) {
    if (error.response) {
      // The request was made and the server responded with a status code
      // that falls out of the range of 2xx
      return error.response.data;
      console.log(error.response.data);
      console.log(error.response.status);
      console.log(error.response.headers);
    } else {
      // Something happened in setting up the request that triggered an Error
      return error.message;
      console.log("Error", error.message);
    }
  }
};
