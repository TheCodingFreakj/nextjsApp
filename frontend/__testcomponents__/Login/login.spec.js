import React from "react";
import { mount, shallow } from "enzyme";
import axios from "axios";
import SigninComp from "../../components/SigninComponent";
import { signin } from "../../actions/auth";

describe("<Login /> with no props", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SigninComp />);
  });

  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("has a login input button", () => {
    expect(
      wrapper.containsMatchingElement(
        <input type="submit" className="btn btn-primary " value="Signin" />
      )
    ).toBeTruthy();
  });

  it("has a email input field", () => {
    expect(
      wrapper.containsMatchingElement(<input type="email" />)
    ).toBeTruthy();
  });

  it("has a password input field", () => {
    expect(
      wrapper.containsMatchingElement(<input type="password" />)
    ).toBeTruthy();
  });

  it("renders a email input", () => {
    expect(wrapper.find('input[type="email"]').length).toEqual(1);
  });
  it("renders a password input", () => {
    expect(wrapper.find('input[type="password"]').length).toEqual(1);
  });

  it("renders a submit input", () => {
    expect(wrapper.find('input[type="submit"]').length).toEqual(1);
  });

  it("should respond to change event and change the state of the Login Component", () => {
    wrapper.find('input[type="email"]').simulate("change", {
      target: { name: "email", value: "pallavidapriya75@gmail.com" },
    });
    expect(wrapper.find('input[type="email"]').prop("value")).toEqual(
      "pallavidapriya75@gmail.com"
    );
  });

  it("should respond to change event and change the state of the Login Component", () => {
    wrapper.find('input[type="password"]').simulate("change", {
      target: { name: "password", value: "mypassword" },
    });

    expect(wrapper.find('input[type="password"]').prop("value")).toEqual(
      "mypassword"
    );
  });

  it("when the form is submiited the event is cancelled", () => {
    wrapper = shallow(<SigninComp signupForm={() => {}} />);
    let prevented = false;
    wrapper.find("form").simulate("submit", {
      preventDefault: () => {
        prevented = true;
      },
    });
    expect(prevented).toBeTruthy();
  });

  test("Should call onSubmit prop for valid form submission", () => {
    const data = {
      email: "pallavidapriya75@gmail.com",
      password: "mypassword",
    };
    const onSubmitSpy = jest.fn();
    const formEventMocked = { preventDefault: jest.fn() };
    //the data that is passed in to the form through the data variable
    //then passed to the onSubmit function when the form is submitted
    wrapper = shallow(<SigninComp data={data} onSubmit={() => onSubmitSpy} />);
    wrapper.find("form").simulate("submit", formEventMocked);
    setTimeout(() => {
      expect(onSubmitSpy).toBeCalledTimes(1);
      expect(onSubmitSpy).toHaveBeenLastCalledWith({
        email: data.email,
        password: data.password,
      });
      done();
    }, 0);
  });
});
