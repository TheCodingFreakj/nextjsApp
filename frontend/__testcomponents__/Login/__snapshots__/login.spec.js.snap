// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> with no props should match the snapshot 1`] = `"<form class=\\"text-center\\"><div class=\\"form-group\\"><input type=\\"email\\" class=\\"form-control\\" placeholder=\\"Your Email\\" value=\\"\\" required=\\"\\"/></div><div class=\\"form-group\\"><input type=\\"password\\" class=\\"form-control\\" placeholder=\\"Your Password\\" value=\\"\\" required=\\"\\"/></div><div><input type=\\"submit\\" class=\\"btn btn-primary \\" value=\\"Signin\\"/></div></form>"`;
